platform/node/index.d.ts

```diff
@@ -0,0 +1,49 @@
+declare module '@maplibre/maplibre-gl-native' {
+  enum Resource {
+    Unknown = 0,
+    Style = 1,
+    Source = 2,
+    Tile = 3,
+    Glyphs = 4,
+    SpriteImage = 5,
+    SpriteJSON = 6,
+  }
+
+  type RequestResponse = {
+    data: ArrayBuffer;
+    modified?: Date;
+    expires?: Date;
+    etag?: string;
+  };
+
+  type MapOptions = {
+    request: (
+      request: { url: string; kind: Resource },
+      callback: (error?: Error, response?: RequestResponse) => void,
+    ) => void;
+    ratio?: number;
```

Comment:
shouldn't '[mode](https://github.com/maplibre/maplibre-gl-native/blob/main/platform/node/src/node_map.cpp#L1402-L1409)' also be in map options?

Reply:
I'm not really good with C – possible enum values can be `tile` or `static`?

Reply:
ya, setting mode to 'tile' puts it into tile mode. I think anything else does static mode, but I usually do put 'static'

Reply:
What's the difference between the modes? Are their any docs/examples?

Reply:
From a quick test I assume only a single tile will be fetched and rendered.

Reply:
From what I have read here, the purpose of tile mode is to return a single tile. However, from what I learned in https://github.com/maplibre/maplibre-gl-native/issues/284 it seems to help with labels that cross tile borders. 

For example, in tile mode it uses more padding for collision index ( https://github.com/maplibre/maplibre-gl-native/blob/main/src/mbgl/text/collision_index.cpp#L26-L50 ), so it is able to consider labels from the next tiles (at least that's my understanding)